test/generators/UserAnswersGenerator.scala

```diff
@@ -47,6 +48,7 @@ trait UserAnswersGenerator extends TryValues {
         internalId = id,
         identifier = utr,
         sessionId = sessionId,
+        newId = newId,
```

Comment:
this is wrong, maybe it is working in tests but it is not doing the same logic as in repository, you are suppose to use combination of id and utr and sessionId

Reply:
done